pkg/api/handlers/mcpcatalogs.go

```diff
@@ -72,78 +72,38 @@ func (h *MCPCatalogHandler) Refresh(req api.Context) error {
 	return req.Update(&catalog)
 }
 
-// Create creates a new catalog.
-func (h *MCPCatalogHandler) Create(req api.Context) error {
+// Update updates a catalog (admin only, default catalog only).
+func (h *MCPCatalogHandler) Update(req api.Context) error {
 	var manifest types.MCPCatalogManifest
 	if err := req.Read(&manifest); err != nil {
 		return fmt.Errorf("failed to read catalog manifest: %w", err)
 	}
 
-	// Validate the URLs
-	for _, urlStr := range manifest.SourceURLs {
-		u, err := url.Parse(urlStr)
-		if err != nil {
-			return types.NewErrBadRequest("invalid URL: %v", err)
-		}
-
-		if u.Scheme != "https" {
-			return types.NewErrBadRequest("only HTTPS URLs are supported")
-		}
-	}
-
-	catalog := v1.MCPCatalog{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: system.CatalogPrefix,
-			Namespace:    req.Namespace(),
-		},
-		Spec: v1.MCPCatalogSpec{
-			DisplayName:    manifest.DisplayName,
-			SourceURLs:     manifest.SourceURLs,
-			AllowedUserIDs: manifest.AllowedUserIDs,
-		},
-	}
-
-	if err := req.Create(&catalog); err != nil {
-		return fmt.Errorf("failed to create catalog: %w", err)
-	}
-
-	return req.Write(convertMCPCatalog(catalog))
-}
-
-// Update updates a catalog.
-func (h *MCPCatalogHandler) Update(req api.Context) error {
-	var manifest types.MCPCatalogManifest
-	if err := req.Read(&manifest); err != nil {
-		return fmt.Errorf("failed to read catalog manifest: %w", err)
+	catalogID := req.PathValue("catalog_id")
+	if catalogID != system.DefaultCatalog {
+		return types.NewErrBadRequest("only the default catalog can be updated")
 	}
 
 	var catalog v1.MCPCatalog
-	if err := req.Get(&catalog, req.PathValue("catalog_id")); err != nil {
+	if err := req.Get(&catalog, catalogID); err != nil {
 		return fmt.Errorf("failed to get catalog: %w", err)
 	}
 
-	if catalog.Spec.IsReadOnly {
-		return types.NewErrBadRequest("cannot update a read-only catalog")
-	}
-
-	if manifest.DisplayName == "" {
-		return types.NewErrBadRequest("display name is required")
-	}
-
+	// The only field that can be updated is the source URLs.
 	for _, urlStr := range manifest.SourceURLs {
-		u, err := url.Parse(urlStr)
-		if err != nil {
-			return types.NewErrBadRequest("invalid URL: %v", err)
-		}
+		if urlStr != "" {
+			u, err := url.Parse(urlStr)
+			if err != nil {
+				return types.NewErrBadRequest("invalid URL: %v", err)
+			}
 
-		if u.Scheme != "https" {
-			return types.NewErrBadRequest("only HTTPS URLs are supported")
+			if u.Scheme != "https" {
+				return types.NewErrBadRequest("only HTTPS URLs are supported")
+			}
```

Comment:
The default catalog in the Docker container will have a `./catalog` as the `urlStr`, which will fail this check. Since we only have one catalog, I could see a user adding multiple `SourceURLs` to it. However, they would only be able to do that if they first removed `./catalog`.

Reply:
Good catch